trie/committer.go

```diff
@@ -30,28 +31,44 @@ type committer struct {
 	nodes       *trienode.NodeSet
 	tracer      *tracer
 	collectLeaf bool
+	parallel    bool
 }
 
 // newCommitter creates a new committer or picks one from the pool.
-func newCommitter(nodeset *trienode.NodeSet, tracer *tracer, collectLeaf bool) *committer {
+func newCommitter(nodes *trienode.NodeSet, tracer *tracer, collectLeaf bool, parallel bool) *committer {
 	return &committer{
-		nodes:       nodeset,
+		nodes:       nodes,
 		tracer:      tracer,
 		collectLeaf: collectLeaf,
+		parallel:    parallel,
 	}
 }
 
+type wrapNode struct {
```

Comment:
The whole concept of wrapping nodes -- I don't see the point in it. Why is that needed? Couldn't you just copy the path for each goroutine, and then let each goroutine work on it's own path-copy individually without risking any cross-goroutine disruptions?

Reply:
Because multiple goroutines calling `AddNode` will cause concurrent map writes.
This could be solved by using mutex, but using wrapping nodes to avoid lock/unlock can save a little more time.

Reply:
Well, if we're parallelizing things, you could have a `chan`where you send off the nodes, and then have a dedicated goroutine which just reads the chan and invokes `AddNode` sequentially. If you make the chan use some buffering, then it should be faster than using a lock.

Reply:
Yes I did try this approach before. However, it still requires a new struct, similar to `WrapNode`, to initialize the `chan` since `AddNode` requires both the node itself and the path. Additionally, we would need an extra mechanism to monitor when the send operation is completed. Here's the [old draft PR ](https://github.com/ethereum/go-ethereum/compare/master...stevemilk:go-ethereum:committer2)illustrating this approach, rough version.

For the idea  of parallelizing committer and invoking `AddNode` sequentially, current PR can also achieve in a cleaner way, meanwhile just as fast.  I believe both approaches are valid and can work well, and I'm open to further discussion if you think there's a strong case for the alternative.